maat-court-data-api/src/main/java/gov/uk/courtdata/contribution/controller/ContributionsController.java

```diff
@@ -113,4 +117,32 @@ public ResponseEntity<ContributionsDTO> create(@Valid @RequestBody CreateContrib
         createContributionsValidator.validate(createContributions);
         return ResponseEntity.ok(contributionsService.create(createContributions));
     }
+
+    @RequestMapping(value = "/{repId}",
+            method = {RequestMethod.HEAD},
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    @Operation(description = "Retrieve contribution count")
+    @ApiResponse(responseCode = "200",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE)
+    )
+    @ApiResponse(responseCode = "400",
+            description = "Bad Request.",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
+                    schema = @Schema(implementation = ErrorDTO.class)
+            )
+    )
+    @ApiResponse(responseCode = "500",
+            description = "Server Error.",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON_VALUE,
+                    schema = @Schema(implementation = ErrorDTO.class)
+            )
+    )
+    public ResponseEntity<Object> getContributionCount(@PathVariable int repId) {
+        log.info("Get contribution count");
+        maatIdValidator.validate(repId);
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.setContentLength(contributionsService.getContributionCount(repId));
```

Comment:
Why are we storing the number of contributions in the header field **ContentLength** ? The purpose of this header field is to store the size of the content, not the number.
I think we should be returning the record count in the body of the HTTP response, possibly as a named field in a JSON structure. 

> The Content-Length entity-header field indicates the size of the entity-body, in decimal number of OCTETs, sent to the recipient or, in the case of the HEAD method, the size of the entity-body that would have been sent had the request been a GET.

https://stackoverflow.com/questions/2773396/whats-the-content-length-field-in-http-header

Reply:
The head  request is standard pattern that use with in the application to get count. Also discussed this with @Mhowell494  and @AafAnsari and I have kept header implementation separately for code clarity